Tests/Factory/Assembly/TestAssemblies/TyphoonConfigAssembly.m

```diff
@@ -30,4 +32,13 @@ - (id)anotherKnight
     }];
 }
 
+- (id)blockKnight
+{
+    return [TyphoonBlockDefinition withClass:[Knight class] block:^id{
```

Comment:
Yes. We need to know the component type beforehand so Typhoon would be able to use this definition for auto-injections and injections by property type.